configuration/exodus-pipeline.yaml

```diff
@@ -438,10 +477,113 @@ Resources:
+        ComputeType: BUILD_GENERAL1_SMALL
+        Image: aws/codebuild/amazonlinux2-x86_64-standard:3.0
+      Source:
+        Type: CODEPIPELINE
+        BuildSpec: |
+          version: 0.2
+          phases:
+            install:
+              runtime-versions:
+                python: 3.7
+              commands:
+                - pip install --upgrade pip
+                - pip install tox
+            build:
+              commands:
+                - git clone https://github.com/$REPO_OWNER/$REPO_NAME.git -b $REPO_BRANCH
+                - cd ./$REPO_NAME
+                - virtualenv integration_tests
+                - source integration_tests/bin/activate
+                - tox -e py37 ./tests/integration -- --cdn-test-url https://$(aws cloudformation describe-stacks --stack-name $STACK_NAME | jq -r '.Stacks[0].Outputs[] | select(.OutputKey=="Distribution").OutputValue')
+      Artifacts:
+        Type: CODEPIPELINE
+      TimeoutInMinutes: 15
+
+  RollbackLambdasCodeBuild:
+    Type: AWS::CodeBuild::Project
+    # Create CodeBuild project for running integration tests against the dev pipeline
+    Condition: NotDev
+    Properties:
+      Name: !Sub ${project}-rollback-${env}
+      ServiceRole: !GetAtt CodeBuildRole.Arn
+      Environment:
+        Type: LINUX_CONTAINER
+        ComputeType: BUILD_GENERAL1_SMALL
+        Image: aws/codebuild/amazonlinux2-x86_64-standard:2.0
+        EnvironmentVariables:
+          - Name: PROJECT
+            Type: PLAINTEXT
+            Value: !Sub ${project}
+          - Name: ENV_TYPE
+            Type: PLAINTEXT
+            Value: !Sub ${env}
+          - Name: OAI
+            Type: PLAINTEXT
+            Value: !Sub ${oai}
+      Source:
+        Type: NO_SOURCE
+        BuildSpec: |
+          version: 0.2
+          phases:
+            build:
+              commands:
+                - aws s3 cp s3://$PROJECT-pipeline-artifacts/build-$ENV_TYPE .
+                - unzip build-$ENV_TYPE
+                - aws cloudformation deploy --template-file exodus-lambda-pkg.yaml --capabilities CAPABILITY_NAMED_IAM --stack-name $PROJECT-lambda-$ENV_TYPE --parameter-overrides env=$ENV_TYPE project=$PROJECT oai=$OAI
+      Artifacts:
+        Type: NO_ARTIFACTS
+      TimeoutInMinutes: 10
+
+  RollbackEventRule:
+    Type: AWS::Events::Rule
+    # Create event rule for rollback
+    Condition: NotDev
+    Properties:
+      EventPattern: !Sub '{
+          "resources": [
+            "arn:aws:codepipeline:${region}:${AWS::AccountId}:${CodePipeline}"
+          ],
+          "source": [
+            "aws.codepipeline"
+          ],
+          "detail-type": [
+            "CodePipeline Action Execution State Change"
+          ],
+          "detail": {
+            "state": [
+              "FAILED"
+            ],
+            "stage": [
+              "Test"
+            ],
+            "action": [
+              "IntegrationTests"
+            ]
+          }
+        }'
```

Comment:
I don't think it's necessary to inject JSON here, YAML formatting should work fine.
> resources:
     - !Sub arn:aws:codepipeline:${region}:${AWS:AccountId}:${CodePipeline}